server/tests/rating.test.js

```diff
@@ -0,0 +1,90 @@
+    const res = await router().patch('/api/ratings/1')
+      .set('token', token).send({ stars: 4 });
+    expect(res.body).to.be.an('object');
+    expect(res.body.status).to.equals(200);
+    expect(res.body.message).to.be.a('string');
+    expect(res.body.data).to.be.an('object');
+    expect(res.body.data.rate).to.be.a('string');
+  });
+  it('should rate the accommodation with no problems', async () => {
+    const res = await router().patch('/api/ratings/1')
+      .set('token', token).send({ stars: 3 });
+    expect(res.body).to.be.an('object');
+    expect(res.body.status).to.equals(200);
+    expect(res.body.message).to.be.a('string');
+    expect(res.body.data).to.be.an('object');
+    expect(res.body.data.rate).to.be.a('string');
+  });
+  it('should rate the accommodation with no problems', async () => {
+    const res = await router().patch('/api/ratings/1')
+      .set('token', token).send({ stars: 2 });
+    expect(res.body).to.be.an('object');
+    expect(res.body.status).to.equals(200);
+    expect(res.body.message).to.be.a('string');
+    expect(res.body.data).to.be.an('object');
+    expect(res.body.data.rate).to.be.a('string');
+  });
+  it('should rate the accommodation with no problems', async () => {
+    const res = await router().patch('/api/ratings/1')
+      .set('token', token).send({ stars: 1 });
+    expect(res.body).to.be.an('object');
+    expect(res.body.status).to.equals(200);
+    expect(res.body.message).to.be.a('string');
+    expect(res.body.data).to.be.an('object');
+    expect(res.body.data.rate).to.be.a('string');
+  });
+  it('should not rate the accommodation with unknown accommodation id', async () => {
+    const res = await router().patch('/api/ratings/145447')
+      .set('token', token).send({ stars: 4 });
+    expect(res.body).to.be.an('object');
+    expect(res.body.status).to.equals(400);
+    expect(res.body.error).to.be.a('string');
+    expect(res.body.error).to.equals('Please provide a authentic accomodation');
+  });
+  it('should not rate the accommodation with no token', async () => {
+    const res = await router().patch('/api/ratings/145447')
+      .send({ stars: 4 });
+    expect(res.body).to.be.an('object');
+    expect(res.body.status).to.equals(401);
+    expect(res.body.error).to.be.a('string');
+    expect(res.body.error).to.equals('Please provide a token first');
+  });
+  it('should not rate the accommodation with unknown user', async () => {
+    const res = await router().patch('/api/ratings/1')
+      .set('token', invalid).send({ stars: 4 });
+    expect(res.body).to.be.an('object');
+    expect(res.body.status).to.equals(401);
+    expect(res.body.error).to.be.a('string');
+    expect(res.body.error).to.equals('Sign up first please');
+  });
+  it('should rate the acc', async () => {
+    const res = await router().patch('/api/ratings/145447')
+      .set('token', `${otherToken.UnExistUser}gfsdfgdfg`).send({ stars: 4 });
```

Comment:
What does ( gfsdfgdfg ) mean ?

Reply:
Here I tried to simulate an invalid token. I will store this into a variable

Reply:
Give it a proper name.

Reply:
Ok It's done now